docs/docker/ocpdiag.docker

```diff
@@ -0,0 +1,38 @@
+FROM centos:latest
+
+###############################################################
+# 
+# Sample docker build file to experiment with ocpdiag
+#
+# paste this in ocpdiag.docker
+#
+# docker build --rm ocpdiag.docker centos:ocpdiag
+# 
+# docker run -it --rm centos:ocpdiag
+#
+###############################################################
+
+RUN sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
+RUN sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
+RUN dnf install -y dnf-plugins-core
+RUN dnf copr enable -y vbatts/bazel
+RUN dnf install -y gcc
+RUN dnf install -y gcc-c++
+RUN dnf install -y unzip
+RUN dnf install -y bazel4
+RUN dnf install -y python3
+RUN dnf install -y go
+RUN python3 -m pip install dataclasses
+
+COPY ocp-diag-core-main.zip /
+RUN cd / && unzip ocp-diag-core-main.zip
+
+WORKDIR /ocp-diag-core-main
+RUN python3 ./apis/python/examples/demo.py 2>&1 > /ocpdiag_py_demo.log
```

Comment:
this seems like a rather odd usage of run commands; these are meant to _build_ the image, but i get the feeling youre using them here as a demo (which also answers my question above). so by doing this youre baking in the results of these commands into the image, which has various implications.

what i think you should do is to add all of this to a shell script, pull that into the image and then use it as entrypoint (which may end up in the interactive shell if that's a goal)
this way, the image stays clean of artifacts that variate (like these log outputs) and docker can reuse it without rebuilding every time

Reply:
Your prompt review of this beat me to setting up context, hopefully my explanation on the intent in an earlier comment explains how I approached this and why I put it out here as a draft.

So, yes, those executions are very much not intended as part of the build file and more of an example so others can reproduce the build error. 

If deemed worthwhile I could see the validator become an image and the run of it could be parametrized to take test output as input

Reply:
let's revisit this after @dhawkes tells us about the dev docker file